packages/protocol/contracts/layer1/shasta/impl/ShastaForcedInclusionStore.sol

```diff
@@ -7,23 +7,22 @@ import { LibAddress } from "src/shared/libs/LibAddress.sol";
 import { LibBlobs } from "../libs/LibBlobs.sol";
 import { LibMath } from "src/shared/libs/LibMath.sol";
 
-/// @title ForcedInclusionStore2
-/// @dev A contract for storing and managing forced inclusion requests. Forced inclusions allow
-/// users to pay a fee to ensure their transactions are included in a block. The contract maintains
-/// a FIFO queue of inclusion requests.
+/// @title ShastaForcedInclusionStore
+/// @dev Abstract contract for storing and managing forced inclusion requests. Forced inclusions
+/// allow users to pay a fee to ensure their transactions are included in a block. The contract
+/// maintains a FIFO queue of inclusion requests.
 /// @dev Inclusion delay is measured in seconds, since we don't have an easy way to get batch number
 /// in the Shasta design.
 /// @dev We only allow one forced inclusion per L1 transaction to avoid spamming the proposer.
 /// @dev Forced inclusions are limited to 1 blob only, and one L2 block only(this and other protocol
 /// constrains are enforced by the node and verified by the prover)
 /// @custom:security-contact security@taiko.xyz
-contract ForcedInclusionStore2 is EssentialContract, IForcedInclusionStore {
+abstract contract ShastaForcedInclusionStore is EssentialContract, IForcedInclusionStore {
```

Comment:
To minimise the Inbox code size, we can use a library to capture the logics then in Inbox.sol, we use the library's functions.